### ISSUE TITLE
support the Native-UTF8-On-Windows version of R

### ISSUE THREAD

**Issue by shrektan** [rstudio/rstudio#8461]
### Description

Hi, I'm currently testing [the experimental version of R on Windows, which uses UTF-8 as the native encoding](https://www.r-project.org/nosvn/winutf8/R-devel-win.exe) thus may solve most of the encoding related headaches from the root...

**Unfortunately, RStudio IDE currently can't work with this version of R.**

What's worse is, I can't switch back to the normal version of R, at least not by clicking the "Tools=>Options" menu, as there's no response any longer...

### Screenshot

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/8368933/99996097-418e9480-2df6-11eb-9eba-21bec7c8f637.png">

### Request

So my feature request is : **RStudio IDE should be prepared for this "future" version of R on Windows**

Thanks.

### References

- [Windows/UTF-8 Build of R and CRAN Packages, Tomas Kalibera](https://developer.r-project.org/Blog/public/2020/07/30/windows/utf-8-build-of-r-and-cran-packages/)

- [UTF-8 Support on Windows, Tomas Kalibera](https://developer.r-project.org/Blog/public/2020/05/02/utf-8-support-on-windows/)

- [R-dev/WindowsBuilds/winutf8](https://svn.r-project.org/R-dev-web/trunk/WindowsBuilds/winutf8/winutf8.html)

**Comment by kevinushey** [comment]
FWIW, one escape hatch is to hold down the `Ctrl` key when launching RStudio. You'll then be presented with a list of R installations to use during launch.

**Comment by kevinushey** [comment]
I took a look and this is caused by an error when we try to invoke `cmdlineoptions` here:

https://github.com/rstudio/rstudio/blob/321b731511ddf7b58b0f5cb322d9106defba4285/src/cpp/r/session/REmbeddedWin32.cpp#L120-L125

Here's what I see from WinDbg:

```
0:000> g
WARNING: Continuing a non-continuable exception
(2088.1ea0): Security check failure or stack buffer overrun - code c0000409 (!!! second chance !!!)
ucrtbase!invoke_watson+0x18:
00007ffc`be201218 cd29            int     29h
0:000> k
 # Child-SP          RetAddr           Call Site
00 00000097`ebffa3c0 00007ffc`be1b1dc1 ucrtbase!invoke_watson+0x18
01 00000097`ebffa3f0 00007ffc`be1b1c89 ucrtbase!_invalid_parameter+0x12d
02 00000097`ebffa430 00007ffc`be1e0b92 ucrtbase!invalid_parameter_noinfo+0x9
03 00000097`ebffa470 00007ffc`be19bd2e ucrtbase!_wsetlocale+0x445c2
04 00000097`ebffa4f0 00007ffc`be19bc4f ucrtbase!<lambda_2116bde18c9e5f34230805ea4a4660ed>::operator()+0xae
05 00000097`ebffa570 00007ffc`be19bc07 ucrtbase!__crt_seh_guarded_call<char *>::operator()<<lambda_e9ce07acdb0b138e546925ae9f1a2c9c>,<lambda_2116bde18c9e5f34230805ea4a4660ed> &,<lambda_5d037afbfc54bf1ca80d3d1ee4062886> >+0x3b
06 00000097`ebffa5a0 00000000`6c8303d3 ucrtbase!setlocale+0x47
07 00000097`ebffa5e0 00000000`6c723179 R!R_ReplDLLinit+0x113
08 00000097`ebffa740 00007ff7`e62b64e0 R!cmdlineoptions+0x89
09 00000097`ebffd3d0 00007ff7`e62a882c rsession!rstudio::r::session::runEmbeddedR+0x80 [c:\users\kevinushey\rstudio-v14\src\cpp\r\session\rembeddedwin32.cpp @ 130] 
0a 00000097`ebffd570 00007ff7`e599e7d0 rsession!rstudio::r::session::run+0xf1c [c:\users\kevinushey\rstudio-v14\src\cpp\r\session\rsession.cpp @ 371] 
0b 00000097`ebffda60 00007ff7`e663d128 rsession!main+0x3ab0 [c:\users\kevinushey\rstudio-v14\src\cpp\session\sessionmain.cpp @ 2166] 
0c (Inline Function) --------`-------- rsession!invoke_main+0x22 [d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 78] 
0d 00000097`ebfff930 00007ffc`bf7c7034 rsession!__scrt_common_main_seh+0x10c [d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 288] 
0e 00000097`ebfff970 00007ffc`c06dcec1 KERNEL32!BaseThreadInitThunk+0x14
0f 00000097`ebfff9a0 00000000`00000000 ntdll!RtlUserThreadStart+0x21
```

We invoke `cmdlineoptions()` because (AFAICR) this is the only way for us to set the memory limit properly:

https://github.com/rstudio/rstudio/issues/4986

**Comment by jmcphers** [comment]
Fixed in Juliet Rose via #8478.
